### PR TITLE
add abstracted support for ipinip/autosnat/vrid

### PR DESCRIPTION
**Pull request by dougwig**

### CODE REVIEW

File: acos_client/v30/axapi_http.py [hunk 1]
```diff
@@ -108,6 +113,36 @@ def request(self, method, api_url, params={}, headers=None,
...
+                break
+            except socket.error as e:
+                # Workaround some bogosity in the API
+                if e.errno in self.retry_errnos:
+                    time.sleep(0.1)
+                    last_e = e
+                    continue
+                raise e
+            except requests.exceptions.ConnectionError as e:
+                if 'BadStatusLine' in str(e):
```
Comment (Cedev) [inline acos_client/v30/axapi_http.py]: I wouldn't bet that the `errno`s only come back in `socket.error`

File: acos_client/v21/axapi_http.py [hunk 2]
```diff
@@ -122,6 +123,10 @@ def __init__(self, host, port=None, protocol="https", client=None, timeout=None)
                 self.port = 80
             else:
                 self.port = 443
+        if retry_errno_list is not None:
+            self.retry_errnos = retry_errno_list
+        else:
+            self.retry_errnos = [errno.ECONNRESET, errno.ECONNREFUSED]
```
Comment (Cedev) [inline acos_client/v21/axapi_http.py]: What if I want to add errors to the existing retry error list without knowing what the "bogosity in the API" is?

**Comment by dougwig** [comment]
check

**Comment by Cedev** [comment]
What is this PR for? The first two changes were merged in #114. Some of the changes don't match the title; they are retry logic changes. I thought we already had ipinip and source nat auto support, but it seems to be being added, not changed.

**Comment by Cedev** [comment]
check

**Comment by dougwig** [comment]
check

**Comment by Cedev** [comment]
Replaced by #123

**Pull request closed without merging**